### PR TITLE
Doc of type hints and minor changes 

### PR DESCRIPTION
**Pull request by tarcisiozf**
* PHPDOC of all methods to have type hints
* Remove cast to object of annotation classes
* Remove deprecated use of __toString()
* minor changes

### CODE REVIEW

File: src/ServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+
+namespace Lipht;
+
+
```
Comment (SparK-Cruz) [inline src/ServiceProvider.php]: These extra 3 empty lines are weird...
```
<?php
namespace Lipht;

```
Is enough wasted space I guess...

File: src/AnnotatedClass.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+namespace Lipht;
+
+
+class AnnotatedClass extends AnnotatedMember
+{
+    /**
+     * @var object
+     */
+    public $methods;
+}
```
Comment (SparK-Cruz) [inline src/AnnotatedClass.php]: New line at EOF is a good practice

File: src/AnnotatedClass.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+
+namespace Lipht;
+
+
```
Comment (SparK-Cruz) [inline src/AnnotatedClass.php]: Too much spacing

File: src/AnnotatedClass.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+class AnnotatedClass extends AnnotatedMember
+{
+    /**
+     * @var object
```
Comment (SparK-Cruz) [inline src/AnnotatedClass.php]: These `@var` annotations could be a one-liner `/** @var object */`

**Comment by tarcisiozf** [comment]
@SparK-Cruz fixed

**Pull request merged**